package/MDAnalysis/analysis/base.py

```diff
@@ -450,6 +487,48 @@ def run(self, start=None, stop=None, step=None, frames=None,
         self._conclude()
         return self
 
+    def _setup_bslices(self, start=None, stop=None, step=None, frames=None):
+        self._bslices = [(start, stop, step, frames)]
+    
+    def _setup_scheduler(self, **kwargs):
+        self._scheduler_kwargs = kwargs
+
+    def run(self, start=None, stop=None, step=None, frames=None,
+            verbose=None, scheduler=None, 
+            *, progressbar_kwargs={},
+            ):
+        if scheduler is None: # fallback to the local scheduler
+            self._compute(start=start, stop=stop, step=step, frames=frames, 
+            verbose=verbose, progressbar_kwargs=progressbar_kwargs)
+        else:
+            self._setup_scheduler()
+            if scheduler == 'localdask': # imitation of dask mainly for testing purposes
+                delayed = localdelayed
+            else: # elif scheduler is a type of dask scheduler
+                try:
+                    from dask.delayed import delayed
+                except ImportError:
+                    def delayed(*args, **kwargs): # need implementation to avoid syntax warnings below
+                        pass
+                    # raising exception here allows me to skip potentionally time-consuming setup steps
+                    raise ImportError('Please install dask for this functionality')
+            self._setup_bslices(start=start, stop=stop, step=step,
+                frames=frames)
+            computations = delayed(
+                [
+                    delayed(self.run)(start=bstart, stop=bstop, step=bstep, frames=bframes)
```

Comment:
This part is getting difficult to follow with all other comments, but it's `self.run` or `self._compute`?